duckduckgo/duckduckgo-privacy-extension
# Browserify background processes

**Reviewer:** @dharb @jdorweiler 

**CC:** @russellholt 
**Depends on:**
Merged: https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/141 https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/142 https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/143 https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/148 https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/149 https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/155 https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/152

## Description:

This PR converts the background processes to using Browserify. It's a collection of multiple different PRs (linked above) so you can look at those if you want to get a piece-by-piece of what this PR contains.

Also see https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/156/files?w=1 for a cleaner diff since I've changed indentation in a few places.

## Steps to test this PR:

1. Run unit tests and verify everything passes
2. Visit a lot of sites and make sure all features of the extension work and there's no JS errors (that weren't there before).

Sorry about the vagueness. Given I've not changed the logic behind the individual components, I think any breakage that may arise out this would be pretty noticeable (e.g. no trackers being clocked at all as opposed to the number of trackers being off by 2).

## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
